CDCgov/dibbs-site
# Incorporate classnames package and do some general cleanup

## Summary

This PR does the following:

- Adds and makes use of `classNames` where appropriate
- Adds some reasonable defaults for `prettier` and `eslint`
  - A couple of new `devDependencies` were added for this 
- Formats all applicable files using the new `prettier` defaults
- Ensures the `Footer` component renders a `footer` element rather than another `header`
- Moves `NavigationLink` and its prop type into its own file
- Moved the heavily used `basePath` variable into its own `constants.ts` file
- Fixed an error where Next.js couldn't find the `favicon.ico`
- Some other minor cleanup 

## Testing

1. Run `npm i` to install the new packages
2. Start the project (restart might be needed)
3. Make sure everything generally still works and looks the same

If you happen to run into any issues, it might be worth trying to delete the `.next` folder and restarting your dev build.

## Discussion

The "reasonable defaults" felt like a good starting point to me but we can definitely adjust these over time if we think any of the ESLint rules aren't useful to us.
